QuizApp/App/Screens/Quiz/QuizListViewModel.swift

```diff
@@ -14,14 +15,42 @@ final class QuizListViewModel {
 
 extension QuizListViewModel {
 
-    func fetchQuiz() {
+    func fetchAllQuizes() {
         Task(priority: .background) {
             let quizes = try await quizUseCase.fetchQuizes()
 
             DispatchQueue.main.async { [weak self] in
                 guard let self = self else { return }
                 self.quizes = quizes
                     .map { QuizViewModel(from: $0) }
+                self.categories = quizes
+                    .map { CategoryViewModel(from: $0.category) }
+//                    .unique()
+            }
+        }
+    }
+
+    func fetchQuiz(for category: String) {
+        Task(priority: .background) {
```

Comment:
You are using Task priority wrong. These network requests are preventing the users from using the app, so they should definitely not be of .`background` priority. Research the docs a bit more and fix throughout the app.